client/src/components/Production/ProductionModal.jsx

```diff
@@ -0,0 +1,151 @@
+                    <option>Complete</option>
+                    <option>Idle</option>
+                    <option>Cancelled</option>
+                  </Fragment>
+                </Select>
+              </FormControl>
+              <FormControl isRequired style={elementStyle}>
+                <FormLabel>Name</FormLabel>
+                <Input ref={nameRef} />
+              </FormControl>
+              <FormControl isRequired style={elementStyle}>
+                <FormLabel>Description</FormLabel>
+                <Input ref={descriptionRef} />
+              </FormControl>
+            </div>
+            {form === 'Part' ? (
+              <ProductionModalPart ref={partRefs} />
+            ) : (
+              <ProductionModalBike ref={bikeRefs} />
+            )}
+            <div style={formStyle}>
+              <FormControl isRequired style={elementStyle}>
+                <FormLabel>Color</FormLabel>
+                <Input ref={colorRef} />
+              </FormControl>
+              <FormControl isRequired style={elementStyle}>
+                <FormLabel>Quantity</FormLabel>
+                <NumberInput min={1} defaultValue={1}>
+                  <NumberInputField ref={quantityRef} />
+                  <NumberInputStepper>
+                    <NumberIncrementStepper />
+                    <NumberDecrementStepper />
+                  </NumberInputStepper>
+                </NumberInput>
+              </FormControl>
+              <FormControl isRequired style={elementStyle}>
+                <FormLabel>Profit Margin</FormLabel>
+                <NumberInput min={1.1} defaultValue={1.1}>
+                  <NumberInputField ref={profitMarginRef} />
+                  <NumberInputStepper>
+                    <NumberIncrementStepper />
+                    <NumberDecrementStepper />
+                  </NumberInputStepper>
+                </NumberInput>
+              </FormControl>
+            </div>
+            <div style={formStyle}>
+              <FormControl isRequired style={elementStyle}>
+                <FormLabel>Start Date</FormLabel>
+                <TextField onChange={handleStartDateInput} type="date" value={startDate} />
+              </FormControl>
+              <FormControl isRequired style={elementStyle}>
+                <FormLabel>End Date</FormLabel>
+                <TextField onChange={handleEndDateInput} type="date" value={endDate} />
+              </FormControl>
+            </div>
+            <FormControl isRequired style={elementStyle}>
+              <FormLabel>Assembly Machine</FormLabel>
+              <Input ref={assemblyMachineRef} />
+            </FormControl>
+            <FormControl style={elementStyle}>
+              <FormLabel>Note</FormLabel>
+              <Textarea placeholder="Additional Note" ref={noteRef} />
+            </FormControl>
+          </ModalBody>
+          <ModalFooter>
+            <Button colorScheme="blue" mr={3} onClick={handleSubmit}>
```

Comment:
should you add conditions to disable the form to prevent empty form leading to 400 error

Reply:
done! @DavidLiang01 let me know what you think of it.